blackboxopt/optimizers/testing.py

```diff
@@ -218,9 +237,48 @@ def raises_evaluation_error_when_reporting_unknown_objective(
     return True
 
 
+def respects_fixed_parameter(
+    optimizer_class: Union[
+        Type[SingleObjectiveOptimizer], Type[MultiObjectiveOptimizer]
+    ],
+    optimizer_kwargs: dict,
+):
+    """Check if optimizer's generated evaluation specifications contain the values
+    a parameter in the search space was fixed to.
+
+    Args:
+        optimizer_class: Optimizer to test.
+        optimizer_kwargs: Expected to contain additional arguments for initializing
+            the optimizer. (`search_space` and `objective(s)` are set automatically
+            by the test.)
+
+    Returns:
+        `True` if the test is passed.
+    """
+    space = ps.ParameterSpace()
+    space.add(ps.ContinuousParameter("my_fixed_param", (-10.0, 200.0)))
```

Comment:
This `ParameterSpace` has actually no changeable parameter anymore, which could lead to all kind of problems for the optimizers. Is this intended, or would it make sense to add another dummy parameter?

Reply:
Depends a bit on how much we rely on users doing reasonable things. Feeding an all fixed space into an optimizer doesn't make too much sense. So proposal: We add a dummy parameter here and add a different test that checks that an all fixed space or a space without parameters raises a ValueError when given to any optimizer - what do you think?

Reply:
That's one option, if we want to cover that. I was mostly concerned about testing with a degenerate case where `len(space) ==0`, which can lead to empty arrays and other nasty bugs that are unrelated to what this test wants to achieve.
In the case that we want to prevent users to accidentally optimizing zero parameters, I agree that the optimizers should throw an error, but I have no strong opinion if that level of sanity checks is required here.